<?xml version="1.0" encoding="UTF-8"?>
<launch>
    
    <arg name="urdf" default="vlp16_bumblebee_vertical" />

    <include file="$(find safer_stack)/launch/spawn/sand_mine.launch"></include>

    <include file="$(find safer_stack)/launch/spawn/spawn_husky.launch">
        <arg name="urdf_extras" value="$(find safer_stack)/urdf/$(arg urdf).urdf.xacro" />
    </include>
    
    <node name="convert_pc_lidar" pkg="safer_stack" type="tf_pointcloud.py" output="screen">
        <remap from="input" to="/lidar/points2"/>
        <remap from="output" to="/lidar/points2_transformed"/>
    </node> 

    <rosparam command="load" file="$(find safer_stack)/config/detection.yaml" />
    <node name='distance_lidar' pkg="safer_stack" type="distance_lidar_nomap.py" />
    
    <node pkg="safer_stack" name="interface" type="interface.py" output="screen">
        <remap from="/edge_distance" to="/lidar/edge_distance" />
    </node>
    
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find safer_stack)/rviz/lidar_nomap.rviz" output="screen"/>

</launch>